tests/unit_tests/test_stats.py

```diff
@@ -96,19 +137,66 @@ def test_watt():
     assert d.b == b
     assert len(d) == 2
 
+    # mean value form adapted from
+    # "Prompt-fission-neutron average energy for 238U(n, f ) from
+    # threshold to 200 MeV" Ethvignot et. al.
+    # https://doi.org/10.1016/j.physletb.2003.09.048
+    exp_mean = 3/2 * a + a**2 * b / 4
+
+    # sample Watt distribution
+    n_samples = 1_000_000
+    samples = d.sample(n_samples, seed=100)
+    assert samples.mean() == pytest.approx(exp_mean, rel=1e-03)
+
 
 def test_tabular():
-    x = [0.0, 5.0, 7.0]
-    p = [0.1, 0.2, 0.05]
+    x = np.array([0.0, 5.0, 7.0])
+    p = np.array([0.1, 0.2, 0.05])
     d = openmc.stats.Tabular(x, p, 'linear-linear')
     elem = d.to_xml_element('distribution')
 
     d = openmc.stats.Tabular.from_xml_element(elem)
-    assert d.x == x
-    assert d.p == p
+    assert all(d.x == x)
+    assert all(d.p == p)
     assert d.interpolation == 'linear-linear'
     assert len(d) == len(x)
 
+    # test linear-lienar sampling
+    d = openmc.stats.Tabular(x, p)
+
+    # compute the expected value (mean) of the
+    # piecewise pdf
+    mean = 0.0
+    d.normalize()
+    for i in range(1, len(x)):
+        y_min = d.p[i-1]
+        y_max = d.p[i]
+        x_min = d.x[i-1]
+        x_max = d.x[i]
+
+        m = (y_max - y_min) / (x_max - x_min)
+
+        exp_val = (1./3.) * m * (x_max**3 - x_min**3)
+        exp_val += 0.5 * m * x_min * (x_min**2 - x_max**2)
+        exp_val += 0.5 * y_min * (x_max**2 - x_min**2)
+
+        mean += exp_val
```

Comment:
Maybe we should just make this a method on the class?

Reply:
Good call. Done!